Port LicenseHeadersTask to java and remove ant layer (7.x backport)

breskeby opened the pull request:
- backports #65310, #65468,  #65509 & #65513 to 7.x branch

breskeby committed:
Port LicenseHeadersTask to java and remove ant layer (#65310)

This ports Licenseheaders task from using groovies antbuilder to rely on the rat ant task
to use rat directly which allows us to port this completely to java. We also refactored rat to
generate xml instead of plain report file.

- More efficient as plain file generation is based on running an xslt transformation on top
of the xml output (via ant rat in a different thread which results in faster task execution as seen
in the attached profiler report)
- Easier machine readable
- By using rat directly we can avoid parsing the log file for reporting.
- Ported more buildSrc sources from groovy to java which has been blocked from porting by
LicenseHeadersTask

elasticmachine commented:
Pinging @elastic/es-delivery (Team:Delivery)

mark-vieira committed:
Correctly set test runtime classpath on standalone rest tests (#65468)

mark-vieira commented:
@breskeby make sure the changes from https://github.com/elastic/elasticsearch/pull/65468 get incorporated into this backport as well.

breskeby committed:
Fix licenseheaders plugin func test on windows (#65509)

breskeby commented:
note to myself: also integrate https://github.com/elastic/elasticsearch/pull/65513 here

breskeby committed:
Fix dependencies info task (#65513)

- Fix dependencies info task
- fix regex in dependencieInfo task after porting to java
